### PR TITLE
Fix remark memory leak and buffer overflow

### PR DESCRIPTION
**Pull request by sgaragagghu**

### CODE REVIEW

File: client/block.c [hunk 1]
```diff
@@ -2012,13 +2013,18 @@ static int add_remark_bi(struct block_internal* bi, xdag_remark_t strbuf)
 	if(!(bi->flags & BI_REMARK)) {
 		return 0;
 	}
-	char *remark_tmp = xdag_malloc(size);
-	if(remark_tmp == NULL) {
-		xdag_err("xdag_malloc failed, [function add_remark_bi]");
-		return 0;
+	pthread_mutex_lock(&g_remark_mutex);
```
Comment (xrdavies) [inline client/block.c]: What is the purpose of this mutex?

**Comment by jonano614** [comment]
But where was the memory leak?

**Comment by sgaragagghu** [comment]
if two thread want to read the same block that hasn't an already loaded remark, they could load the remark at the same time and allocate memory two times

File: client/block.c (hunk 1 above)
Comment (sgaragagghu) [reply client/block.c]: i did it without mutex now

**Comment by xrdavies** [comment]
Indeed the previous implementation has a memory leak issue. For example, if launch two terminals or rpc calls to get block info of the same block at the same time, it may cause the memory leak.

**Pull request merged**